### ISSUE TITLE
Require syntax document

### ISSUE THREAD

**Issue by deafwolf** [mattn/anko#74]
As the title, the readme is too simple.

**Comment by MarcusE1W** [comment]
Have you noticed the examples in anko/_example/scripts/ ? Maybe that helps a bit?

**Comment by AlekSi** [comment]
Still, language specification would be very useful.

**Comment by MichaelS11** [comment]
Was curious, what would this look like? Is there an example from another language like Go, JavaScript, Python, etc.?

**Comment by AlekSi** [comment]
[Go language specification](https://golang.org/ref/spec) is a good example.

**Comment by MichaelS11** [comment]
Yeah, having language specification would be very nice. I personally would like to see 95%+ code coverage testing first.

**Comment by floren** [comment]
@MichaelS11 It is hard to write tests without knowing that the expected behavior is. I agree, a language spec is important. I'm writing a description for the documentation of my own project, but in order to do it I'm having to dive into the anko codebase because the examples aren't sufficient. Right now I'm trying to figure out if it's possible to expand / append to a slice, and what precisely is the difference between a slice and an array anyway.

**Comment by MichaelS11** [comment]
There is a lots still to do, including examples and language specifications. Anyone who would like to help, good PRs are excepted.

Until more examples are made for slices, I would suggest checking out the tests:

https://github.com/mattn/anko/blob/master/vm/vmArraysAndMaps_test.go

**Comment by floren** [comment]
A BNF syntax would be great, although it's been so long since I've written one that I'm not confident I could pull it off.
